storm-client/test/jvm/org/apache/storm/topology/SimpleWindowPartitionCacheTest.java

```diff
@@ -0,0 +1,232 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.topology;
+
+import org.apache.storm.utils.Utils;
+import org.apache.storm.windowing.persistence.SimpleWindowPartitionCache;
+import org.apache.storm.windowing.persistence.WindowPartitionCache;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.FutureTask;
+
+/**
+ * Unit tests for {@link SimpleWindowPartitionCache}
+ */
+public class SimpleWindowPartitionCacheTest {
+
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBuildInvalid1() throws Exception {
+        SimpleWindowPartitionCache.<Integer, Integer>newBuilder()
+            .maximumSize(0)
+            .build(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBuildInvalid2() throws Exception {
+        SimpleWindowPartitionCache.<Integer, Integer>newBuilder()
+            .maximumSize(-1)
+            .build(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testBuildInvalid3() throws Exception {
+        SimpleWindowPartitionCache.<Integer, Integer>newBuilder()
+            .maximumSize(1)
+            .build(null);
+    }
+
+    @Test
+    public void testBuildOk() throws Exception {
+        SimpleWindowPartitionCache.<Integer, Integer>newBuilder()
+            .maximumSize(1)
+            .removalListener((key, val, removalCause) -> {
+            })
+            .build(key -> key);
+    }
+
+    @Test
+    public void testGet() throws Exception {
+        List<Integer> removed = new ArrayList<>();
+        List<Integer> loaded = new ArrayList<>();
+        SimpleWindowPartitionCache<Integer, Integer> cache =
+            SimpleWindowPartitionCache.<Integer, Integer>newBuilder()
+                .maximumSize(2)
+                .removalListener((key, val, removalCause) -> removed.add(key))
+                .build(key -> {
+                    loaded.add(key);
+                    return key;
+                });
+
+        cache.get(1);
+        cache.get(2);
+        cache.get(3);
+        Assert.assertEquals(Arrays.asList(1, 2, 3), loaded);
+        Assert.assertEquals(Collections.singletonList(2), removed);
```

Comment:
Since 2 is the largest un-pinned entry before 3 is loaded. The cache evicts the largest unpinned entry.